pkg/freeze/containerd/pause_test.go

```diff
@@ -0,0 +1,149 @@
+package containerd
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/containerd/containerd"
+	"google.golang.org/grpc"
+	cri "k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
+	"knative.dev/container-freezer/pkg/daemon"
+)
+
+var (
+	containers []*cri.Container
+	results    []string
+	method     string
+)
+
+type FakeContainerdCri struct {
+	containers []*cri.Container
+	results    []string
+	method     string
+}
+
+// Container creates a CRI container with the given container ID and name
+func Container(id, name string) *cri.Container {
+	return &cri.Container{
+		Id: id,
+		Metadata: &cri.ContainerMetadata{
+			Name: name,
+		},
+	}
+}
+
+func (c *FakeContainerdCri) List(ctx context.Context, conn *grpc.ClientConn, podUID string) (*cri.ListContainersResponse, error) {
+	listContainers := &cri.ListContainersResponse{
+		Containers: c.containers,
+	}
+	containerIDs, err := lookupContainerIDs(listContainers)
```

Comment:
seems weird to be calling a private method in the tests, and I'm not sure exactly what's going on here. What I'd expect is that the tests would set a private field on FakeContainerCri telling it what containers to return, and then assert that Freeze called Pause with the right values. If both Thaw/Freeze _and_ the fake List call lookupContainerIDs that seems very confusing.

Reply:
The only place this is actually used is to test if there's a pod that only has a queue-proxy container ... probably don't even need to include this

Reply:
I'd just explicitly test that Pause was called with the expected containers

Reply:
The issue is that the real type doesn't have a "paused" list, so accessing it from the test was proving a little tricky.

Reply:
ok, think I've got this fixed now